src/main/java/com/yahoo/bullet/bql/parser/ASTBuilder.java

```diff
@@ -553,173 +268,155 @@ protected Node defaultResult() {
-        if (nextResult == null) {
-            throw new UnsupportedOperationException("Not yet implemented");
+        if (aggregate != null || nextResult == null) {
+            throw new UnsupportedOperationException("Not implemented");
         }
-        if (aggregate == null) {
-            return nextResult;
-        }
-        throw new UnsupportedOperationException("Not yet implemented");
+        return nextResult;
     }
 
-    private static SortItem.Ordering getOrderingType(Token token) {
-        if (token.getType() == BQLBaseLexer.DESC) {
-            return SortItem.Ordering.DESCENDING;
+    private ExpressionNode stripParentheses(ExpressionNode expression) {
+        if (expression instanceof ParenthesesExpressionNode) {
+            return ((ParenthesesExpressionNode) expression).getExpression();
         }
-        return SortItem.Ordering.ASCENDING;
-    }
-
-    private GroupOperationType getGroupOperationType(BQLBaseParser.FunctionCallContext context) {
-        String functionName = context.qualifiedName().getText().toUpperCase();
-        return GroupOperationType.valueOf(functionName);
+        return expression;
     }
 
-    private <T> Optional<T> visitIfPresent(ParserRuleContext context, Class<T> clazz) {
-        return Optional.ofNullable(context)
-                .map(this::visit)
-                .map(clazz::cast);
+    private static SortItemNode.Ordering getOrdering(Token token) {
+        return token != null && token.getType() == BQLBaseLexer.DESC ? SortItemNode.Ordering.DESCENDING :
+                                                                       SortItemNode.Ordering.ASCENDING;
     }
 
-    private <T> List<T> visit(List<? extends ParserRuleContext> contexts, Class<T> clazz) {
-        return contexts.stream()
-                .map(this::visit)
-                .map(clazz::cast)
-                .collect(toList());
+    private Node visitIfPresent(ParserRuleContext context) {
+        return context != null ? visit(context) : null;
     }
 
-    private static String unquote(String value) {
-        return value.substring(1, value.length() - 1)
-                .replace("''", "'");
+    private <T> List<T> visit(List<? extends ParserRuleContext> contexts, Class<T> type) {
+        return contexts.stream().map(this::visit)
+                                .map(type::cast)
+                                .collect(Collectors.toList());
     }
 
-    private QualifiedName getQualifiedName(BQLBaseParser.QualifiedNameContext context) {
-        List<String> parts = visit(asList(context.identifier()), Identifier.class).stream()
-                .map(Identifier::getValue) // TODO: preserve quotedness
-                .collect(Collectors.toList());
-
-        return QualifiedName.of(parts);
+    private static String unquoteSingle(String value) {
+        return value.substring(1, value.length() - 1).replace("''", "'");
```

Comment:
Why was this in here?

Reply:
String literals use single quotes so we have to remove them. 
The double ' to single ' is a SQL thing. I might have removed that from the acceptable bql though. Not sure. Do you want to keep it or no?

(Example of the double to single: 'doesn''t' is the string doesn't

Reply:
Don't even know why that's a thing. We can ignore it. Of all the things we don't support for SQL, this seems like the least useful thing.